src/app/core/entity-components/entity-list/list-paginator/list-paginator.component.ts

```diff
@@ -0,0 +1,137 @@
+
+  constructor(
+    private sessionService: SessionService,
+    private entityMapperService: EntityMapperService,
+    private router: Router,
+  ) {
+    this.paginatorKey = getUrlWithoutParams(this.router);
+  }
+
+  ngOnInit() {
+    console.log("PaginatorKey: " + this.paginatorKey);
+    this.user = this.sessionService.getCurrentUser();
+    // Use URl as key to save pagination settings
+    this.paginatorPageSize =
+      this.user.paginatorSettingsPageSize[this.paginatorKey] ||
+      this.paginatorPageSize;
+    this.paginatorPageIndex =
+      this.user.paginatorSettingsPageIndex[this.paginatorKey] ||
+      this.paginatorPageIndex;
+    console.log("OnInit aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.showAllToggle =
+      this.paginatorPageSize >= this.dataSource.data.length;
+      console.log("OnChanges aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+    setTimeout(() => {
+      this.paginator.pageIndex = this.paginatorPageIndex;
+      this.paginator.page.next({
+        pageIndex: this.paginator.pageIndex,
+        pageSize: this.paginator.pageSize,
+        length: this.paginator.length,
+      });
+      console.log("AfterViewInit mit Timeout aufgerufen. Length ist " + this.dataSource.data.length);
+    });
+    console.log("AfterViewInit aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  onPaginateChange(event: PageEvent) {
+    this.paginatorPageSize = event.pageSize;
+    this.paginatorPageIndex = event.pageIndex;
+    this.updateUserPaginationSettings();
+    this.showAllToggle =
+      this.paginatorPageSize >= this.dataSource.data.length;
+    console.log("OnPaginateChange aufgerufen. Length ist " + this.dataSource.data.length);
+  }
+
+  getPaginatorPageSizeOptions(): number[] {
+    const ar = [3, 10, 20, 50].filter((n) => {
+      return n < this.dataSource.data.length;
+    });
+    ar.push(this.dataSource.data.length);
+    return ar;
+  }
+
+  getPaginatorPageSize(): number {
+    if (
+      this.dataSource.data.length &&
+      this.paginatorPageSize >= this.dataSource.data.length
+    ) {
+      this.paginatorPageSize = this.dataSource.data.length;
+    }
+    return this.paginatorPageSize;
+  }
+
+  clickShowAllToggle() {
+    if (!this.showAllToggle) {
+      this.paginatorPageSizeBeforeToggle = this.paginatorPageSize;
+      this.paginatorPageSize = this.dataSource.data.length;
+    } else {
+      if (
+        this.paginatorPageSizeBeforeToggle <= this.dataSource.data.length
+      ) {
+        this.paginatorPageSize = this.paginatorPageSizeBeforeToggle;
+      } else {
+        const po = this.getPaginatorPageSizeOptions();
+        this.paginatorPageSize = po.length > 2 ? po[po.length - 2] : po[0];
+      }
+    }
+    this.paginator._changePageSize(this.paginatorPageSize);
+    this.showAllToggle = !this.showAllToggle;
```

Comment:
Doesnt the `[(ngModel)]` of the toggle already change the value of `this.showAllToggle`?

Reply:
I don't really know why, but this manual setting seems to be necessary.